sktime/base/_base.py

```diff
@@ -111,9 +114,126 @@ def clone(self):
         A clone is a different object without shared references, in post-init state.
         This function is equivalent to returning sklearn.clone of self.
         Equal in value to `type(self)(**self.get_params(deep=False))`.
+
+        Returns
+        -------
+        instance of type(self), clone of self (see above)
         """
         return clone(self)
 
+    @classmethod
+    def _get_init_signature(cls):
+        """Get init sigature of cls, for use in parameter inspection.
+
+        Returns
+        -------
+        list of inspect Parameter objects (including defaults)
+
+        Raises
+        ------
+        RuntimeError if cls has varargs in __init__
+        """
+        # fetch the constructor or the original constructor before
+        # deprecation wrapping if any
+        init = getattr(cls.__init__, "deprecated_original", cls.__init__)
+        if init is object.__init__:
+            # No explicit constructor to introspect
+            return []
+
+        # introspect the constructor arguments to find the model parameters
+        # to represent
+        init_signature = inspect.signature(init)
```

Comment:
Thoughts on using inspect.getargspec(init) instead.  From my understanding it returns a named tuple of the args, but also the `varargs` and `keywords` (ie `args` and `kwargs`) which seems like it would make what comes next much easier.... ie:
if len(args) > 0 -> p = args[1:]
  if `varargs` -> raise error

Reply:
Hmm, just kidding, I guess `getfullargspec` is the most up to date version of that.  However, it does seem that signature is generally the preferred method, so I contradict myself, what you initially had is probably fine.

Reply:
> Thoughts on using inspect.getargspec(init) instead?

This is refactored scikit-learn code, and they use `signature`.
I try to be consistent in the content. Anyway, I think it is not an issue, more of a discussion.